torch/_functorch/vmap.py

```diff
@@ -307,6 +309,18 @@ def _register_python_decomposition_vmap(decomp):
 
 
 def vmap_impl(func, in_dims, out_dims, randomness, chunk_size, *args, **kwargs):
+    if has_torch_function(*tree_iter(args)):
+        return handle_torch_function(
```

Comment:
We might need to move vmap_impl/grad_impl to be public APIs, or at least put them in a public module somewhere.

Reply:
If we want that to be a public feature yes I guess so, here they are not obvious to obtain and document.
RE renaming, we could have `vmap_func` (instead of `impl`) which would be a direct call to `vmap` with func and args as inputs.